operate/webapp/src/main/java/io/camunda/operate/webapp/elasticsearch/reader/ElasticsearchListenerReader.java

```diff
@@ -20,11 +41,58 @@
 @Component
 public class ElasticsearchListenerReader extends AbstractReader implements ListenerReader {
 
-  @Autowired private ProcessInstanceReader processInstanceReader;
+  private static final Logger LOGGER = LoggerFactory.getLogger(ElasticsearchListenerReader.class);
+
+  @Autowired private JobTemplate jobTemplate;
+  @Autowired private RestHighLevelClient esClient;
 
   @Override
   public List<ListenerDto> getListenerExecutions(
       final String processInstanceId, final ListenerRequestDto request) {
-    throw new UnsupportedOperationException("Listeners not implemented for Elasticsearch yet.");
+    final TermQueryBuilder processInstanceQ =
+        termQuery(JobTemplate.PROCESS_INSTANCE_KEY, processInstanceId);
+    final TermQueryBuilder flowNodeIdQ =
+        termQuery(JobTemplate.FLOW_NODE_ID, request.getFlowNodeId());
+    final TermQueryBuilder executionListenersQ =
+        termQuery(JobTemplate.JOB_KIND, ListenerType.EXECUTION_LISTENER);
+    final TermQueryBuilder taskListenersQ =
+        termQuery(JobTemplate.JOB_KIND, ListenerType.TASK_LISTENER);
+    final SortBuilder sorting =
+        SortBuilders.fieldSort(request.getSorting().getSortBy())
+            .order(SortOrder.fromString(request.getSorting().getSortOrder()));
+
+    final SearchRequest searchRequest =
+        ElasticsearchUtil.createSearchRequest(jobTemplate, ALL)
+            .source(
+                new SearchSourceBuilder()
+                    .query(
+                        boolQuery()
+                            .must(processInstanceQ)
+                            .must(flowNodeIdQ)
+                            .should(executionListenersQ)
+                            .should(taskListenersQ))
+                    .sort(sorting)
+                    .size(request.getPageSize()));
+
+    final List<JobEntity> jobEntities;
+    try {
+      jobEntities =
+          ElasticsearchUtil.mapSearchHits(
+              esClient.search(searchRequest, RequestOptions.DEFAULT).getHits().getHits(),
+              objectMapper,
+              JobEntity.class);
+
+    } catch (final IOException e) {
+      final String message =
+          String.format(
+              "Exception occurred, while searching for batch operation metadata.", e.getMessage());
```

Comment:
Is this the correct error message? And there is format used but where is the format directive?